crates/ide/src/inlay_hints.rs

```diff
@@ -299,6 +305,26 @@ fn is_argument_similar_to_param_name(
     }
 }
 
+fn is_param_name_similar_to_fn_name(
+    param_name: &str,
+    param_num: usize,
+    fn_name: Option<&String>,
+) -> bool {
+    // if it's the first parameter, don't show it if:
```

Comment:
I feel that this heuristic should only trigger only if there's single parameter.

Reply:
I can imagine myself using an API `apply_to_something(something, this, that)`.
May I ask for any scenarios where this won't declutter (and might hinder ergonomics)?

Reply:
I don't think that the first argument is necessary the biased one. I can imagine APIs like `add_foo(thing_to_add_foo_to, foo)`

Reply:
I'm sorry if I'm misunderstanding, but won't that just display both hints, even with this PR?

Reply:
Yes, and this is the thing I am worrying about -- that we sometimes show, and sometimes hide the hint. I think "single argument" is a rule which covers majority of cases, without introducing subtelties like dependence on the number of paramters.

Reply:
👍